sysutils/AutoRaise/Portfile

```
@@ -0,0 +1,32 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=portfile:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        sbmpost AutoRaise 1.5 v
+
+categories          sysutils
+platforms           macosx
+
+license             {GPL-2 GPL-3}
+maintainers         {@lhaeger gmx.net:lothar.haeger} openmaintainer
+
+description         AutoRaise raises application windows to the front and gives them focus on mouse hover
+long_description    ${description} after a configurable delay. There is also an option to warp the mouse \
+                    to the center of the activated window, using the cmd-tab key combination for example.
+
+homepage            htps://github.com/sbmpost/AutoRaise
+
+checksums           rmd160  6569b47fe2570fe8f079ea06c96c43314a9eb4df \
+                    sha256  a42fe824a3e1f8cb51529b52321ba138bef966a5fa496ac538c466046ced3969 \
+                    size    28613
```

The default file type of the github portgroup is "tarball" which is deprecated. Please use "archive" type instead. This will change the checksums.
```suggestion
github.tarball_from archive

checksums           rmd160  3a343bf8c281f5ea0ed0013ec747b7357d743fed \
                    sha256  19fd38c7e898437b8a310f4039f0d059c657436164f24c0d154a33189dbcfcf0 \
                    size    28566
```

---

Shouldn't we change the default then?

---

> Shouldn't we change the default then?

That will affect a ton of ports.  It's doable, but still a big impact.  About 1,680 ports seem to use the `tarball_from` default right now. We could update them all to set `github.tarball_from` to `tarball`, then change the _portgroup_ default to `archive` right after that. But ideally you would still want to verify that each port still works the same after changing them all.

After doing that, over time as ports get updated to new versions, then you change the source from `tarball` to `archive` (or even `releases`) as you update a port to a new version (or other routine change).

---

There may also be portfiles in third-party repositories using the github portgroup. If we change the defaults of the portgroup that would break those portfiles. An option is to introduce a new github 1.1 portgroup with archive as the default type, and portfiles can migrate to that version of the portgroup over time. However there are many other aspects of the github 1.0 portgroup that I'm no longer delighted with that it might be a good idea to fix at the same time. Even better would be to integrate fetch from github functionality into MacPorts base. I've thought about how best to do this many times over the years but have not arrived at a satisfactory answer. So for now we have what we have.

---

Well, small steps may be better than not moving at all. Do you have a list of the piled-up github issues somewhere? Maybe enough items could be implemented in a v1.1 portgroup with reasonable effort to justify the version bump, leaving the not-so-easy stuff for a later v1.2 or for integration into macports base.